### PR TITLE
ci: prototype running lint / unit tests / coverage as GH actions

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: .github/workflows/unittest_lint.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: "Run lint, unit tests, and cover"
```
Comment (tmatsuo) [inline .github/workflows/unittest_lint.yml:1]: The current name looks good.

Other naming ideas:
"GH actions presubmit"
or
"Lint / Unit Tests / Coverage"
?

**Review by tmatsuo — APPROVED** [review]

File: .github/workflows/unittest_lint.yml (hunk 1 above)
Comment (tseaver) [reply .github/workflows/unittest_lint.yml]: @tmatsuo I've added running `mypy` as well, and retitled it `Lint / Unit Tests / Coverage / Mypy`.

**Comment by tseaver** [comment]
@tmatsuo Do we need to be disabling the tests run via GH Actions in the Kokoro build?

**Comment by tmatsuo** [comment]
@tseaver 
> Do we need to be disabling the tests run via GH Actions in the Kokoro build?

Sorry what do you mean?

**Comment by tseaver** [comment]
@tmatsuo If we are running lint / unit tests / coverage / mypy via GH Actions, should we skip running them on Kokoro, restricting it to only those jobs which need secrets?

**Comment by tmatsuo** [comment]
@tseaver Yes, it makes sense. Tell us which Kokoro job we should disable.
@parthea FYI

**Comment by tseaver** [comment]
@tmatsuo 
>  Yes, it makes sense. Tell us which Kokoro job we should disable.

For this repo, the main `Kokoro` build is completely covered by the GH Actions:  there are no systests or samples being run.  We would also need to work out the spellings to make those actions required in `.kokoro/sync-repo-settings.yaml`.

**Comment by tseaver** [comment]
@parthea I added a `sync_repo_settings.py` here, and configured it to make the GH Actions required.  However, that change isn't made during the PR review process:  it gets applied by the repo automation bot only *after* the PR merges.  Dropping the `Kokoro` job (but not `Kokoro docs-presubmit`) would need to happen in the internal `google3` repo, I think -- while we're at it, we could drop the `Samples` jobs, too -- this repo has none.

**Pull request merged**